versioned/persist/mongodb/src/main/java/org/projectnessie/versioned/persist/mongodb/MongoDatabaseAdapter.java

```diff
@@ -434,27 +612,31 @@ protected boolean doGlobalPointerCas(
 
   @Override
   protected void doCleanUpCommitCas(
-      NonTransactionalOperationContext ctx,
-      Optional<Hash> globalHead,
-      Set<Hash> branchCommits,
-      Set<Hash> newKeyLists,
-      Hash refLogId) {
-    globalHead.ifPresent(h -> client.getGlobalLog().deleteOne(Filters.eq(toId(h))));
-
+      NonTransactionalOperationContext ctx, Set<Hash> branchCommits, Set<Hash> newKeyLists) {
     delete(client.getCommitLog(), branchCommits);
     delete(client.getKeyLists(), newKeyLists);
-    client.getRefLog().deleteOne(Filters.eq(toId(refLogId)));
   }
 
   @Override
-  protected void doCleanUpGlobalLog(
-      NonTransactionalOperationContext ctx, Collection<Hash> globalIds) {
-    delete(client.getGlobalLog(), globalIds);
+  protected void doCleanUpRefLogWrite(NonTransactionalOperationContext ctx, Hash refLogId) {}
+
+  @Override
+  protected ReferenceNames doFetchReferenceNames(
+      NonTransactionalOperationContext ctx, int segment) {
+    Document doc = client.getRefNames().find(Filters.eq(toId(segment))).first();
```

Comment:
Could we do a full scan instead of looping over segments on our side?

Reply:
Updated the code to fetch/resolve up to 25 named references at once.

Reply:
And prefetch reference-names segments (1 by default).

Reply:
Besides will Mongo not batch scan results on the server side (I hope, but do not know)?

Reply:
I suppose we may want to cross-check the ref list table with ref heads as part of the database "maintenance" operation (followup PR). WDYT?

Reply:
Becase ref heads and the ref name list are upgraded separately, so there's a chance the get out of sync?

Reply:
Ah - nah.

When a named-ref's added, the name list is updated first and then the ref-head - so it might happen, that a name exists in the name list but not as a ref-head.
When a named-ref's deleted, the ref head is deleted first, then removed from the names list.

So there's a chance that a name exists in the names list, but not as a ref-head. This isn't a problem, because the named-refs listing only returns those that exist as a ref head as well. For "point queries" with the name, the names list isn't at play.

Reply:
Ah - you wanna show that via a maintenance job. Yea, legit.